records_mover/db/bigquery/bigquery_db_driver.py

```diff
@@ -19,17 +19,23 @@
 
 class BigQueryDBDriver(DBDriver):
     def __init__(self,
-                 db: Union[sqlalchemy.engine.Connection, sqlalchemy.engine.Engine],
+                 db: Optional[Union[sqlalchemy.engine.Connection, sqlalchemy.engine.Engine]],
```

Comment:
why does the BigQueryLoader use the mixin but the BigQueryDriver doesnt?

Reply:
Is it because in the loader itself it might have to open a connection so it cares more about that sort of state?

Reply:
yeah, this is the problem with inconsistent use of hierarchies/inheritance as we mentioned (it would have been nice if all loaders, unloaders, etc. inherited from a base loader class but, here we are)